src/Explorer/Notebook/NotebookContainerClient.ts

```diff
@@ -68,48 +88,58 @@ export class NotebookContainerClient {
 
     const { notebookServerEndpoint, authToken } = this.getNotebookServerConfig();
     try {
-      if (this.checkStatus()) {
-        const response = await fetch(`${notebookServerEndpoint}api/metrics/memory`, {
-          method: "GET",
-          headers: {
-            Authorization: authToken,
-            "content-type": "application/json",
-          },
-        });
-        if (response.ok) {
-          if (this.clearReconnectionAttemptMessage) {
-            this.clearReconnectionAttemptMessage();
-            this.clearReconnectionAttemptMessage = undefined;
-          }
-          const memoryUsageInfo = await response.json();
-          if (memoryUsageInfo) {
-            return {
-              totalKB: memoryUsageInfo.total,
-              freeKB: memoryUsageInfo.free,
-            };
-          }
-        }
-      }
-      return undefined;
+      const runMemoryAsync = async () => {
+        return this._getMemoryAsync(notebookServerEndpoint, authToken);
+      };
+
+      return (async () => {
+        return await promiseRetry(runMemoryAsync, this.retryOptions);
+      })();
```

Comment:
simplefy as 

return promiseRetry(runMemoryAsync, this.retryOptions);

as long as "const memoryUsageInfo = await this.getMemoryUsage();" is awaited, you are fine